src/components/modals/ExcludeClassTimeFilters.tsx

```diff
@@ -0,0 +1,247 @@
+import { storeExcludedClassTimeFilters } from "@/lib/helpers/storage";
+import { ExcludeClassTimeFilter } from "@/types/chain";
+
+export default function ExcludeClassTimeFilters({
+    excludedClassTimeFilters,
+    setExcludedClassTimeFilters,
+}: {
+    excludedClassTimeFilters: ExcludeClassTimeFilter[];
+    setExcludedClassTimeFilters: Dispatch<SetStateAction<ExcludeClassTimeFilter[]>>;
+}) {
+    const theme = useTheme();
+
+    const weekdays = getCapitalizedWeekdays();
+    const [excludedClassTimeWeekday, setExcludedClassTimeWeekday] = useState<number>(1);
+    const [excludedClassTimeStartTime, setExcludedClassTimeStartTime] = useState<DateTime | null>(null);
+    const [excludedClassTimeEndTime, setExcludedClassTimeEndTime] = useState<DateTime | null>(null);
+    const [inputValid, setInputValid] = useState<boolean>(false);
+
+    const validateInput = useCallback(() => {
+        return (
+            excludedClassTimeWeekday >= 1 &&
+            excludedClassTimeWeekday <= 7 &&
+            excludedClassTimeStartTime?.isValid === true &&
+            excludedClassTimeEndTime?.isValid === true &&
+            excludedClassTimeStartTime < excludedClassTimeEndTime &&
+            !excludedClassTimeFilters.some(
+                (it) =>
+                    it.weekday === excludedClassTimeWeekday &&
+                    it.startHour === excludedClassTimeStartTime.hour &&
+                    it.startMinute === excludedClassTimeStartTime.minute &&
+                    it.endHour === excludedClassTimeEndTime.hour &&
+                    it.endMinute === excludedClassTimeEndTime.minute,
+            )
+        );
+    }, [excludedClassTimeFilters, excludedClassTimeWeekday, excludedClassTimeStartTime, excludedClassTimeEndTime]);
+
+    useEffect(() => {
+        setInputValid(validateInput());
+    }, [
+        validateInput,
+        excludedClassTimeFilters,
+        excludedClassTimeWeekday,
+        excludedClassTimeStartTime,
+        excludedClassTimeEndTime,
+    ]);
+
+    const addFilter = () => {
+        if (!validateInput()) {
+            console.log("input was null...");
+            return;
+        }
+
+        setExcludedClassTimeFilters((filters) => {
+            const newFilters = [
+                ...filters,
+                {
+                    weekday: excludedClassTimeWeekday,
+                    startHour: excludedClassTimeStartTime!.hour,
+                    startMinute: excludedClassTimeStartTime!.minute,
+                    endHour: excludedClassTimeEndTime!.hour,
+                    endMinute: excludedClassTimeEndTime!.minute,
+                },
+            ];
+            storeExcludedClassTimeFilters(newFilters);
+            return newFilters;
+        });
+    };
+
+    const removeFilter = (filter: ExcludeClassTimeFilter) => {
+        setExcludedClassTimeFilters((filters) => {
+            const newFilters = filters.filter((it) => it !== filter);
```

Comment:
Not critical, but this equality check is a little scary 😅 Could consider defining a custom equality function.

Reply:
Could consider rewriting it, but I'm not sure what you mean?
To implement a check on each contained value rather than `!===`?

Reply:
Checking each property sound good :+1:

Reply:
:white_check_mark: